src/LuaTransformer.ts

```diff
@@ -5277,23 +5277,28 @@ export class LuaTransformer {
         return luaKeywords.has(name) || luaBuiltins.has(name) || !tsHelper.isValidLuaIdentifier(name);
     }
 
-    protected hasUnsafeSymbolName(symbol: ts.Symbol): boolean {
-        if (luaKeywords.has(symbol.name) || luaBuiltins.has(symbol.name)) {
-            // lua keywords are only unsafe when non-ambient and not exported
-            const isNonAmbient = symbol.declarations.find(d => !tsHelper.isAmbient(d)) !== undefined;
-            return isNonAmbient && !this.isSymbolExported(symbol);
+    protected hasUnsafeSymbolName(symbol: ts.Symbol, tsOriginal?: ts.Identifier): boolean {
+        const isLuaKeyword = luaKeywords.has(symbol.name);
+        const isInvalidIdentifier = !tsHelper.isValidLuaIdentifier(symbol.name);
+        const isAmbient = symbol.declarations.find(d => !tsHelper.isAmbient(d)) === undefined;
```

Comment:
Could be a bit cleaner with `some`:
```suggestion
        const isAmbient = symbol.declarations.some(d => tsHelper.isAmbient(d));
```